cmd/sops/subcommand/updatekeys/updatekeys.go

```diff
@@ -0,0 +1,198 @@
+package updatekeys
+
+import (
+	"fmt"
+
+	"log"
+
+	"path/filepath"
+
+	"os"
+
+	"strings"
+
+	"github.com/fatih/color"
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/cmd/sops/common"
+	"go.mozilla.org/sops/config"
+	"go.mozilla.org/sops/keys"
+	"go.mozilla.org/sops/keyservice"
+)
+
+type Opts struct {
+	InputPath   string
+	GroupQuorum int
+	KeyServices []keyservice.KeyServiceClient
+	Interactive bool
+	ConfigPath  string
+}
+
+func UpdateKeys(opts Opts) error {
+	path, err := filepath.Abs(opts.InputPath)
+	if err != nil {
+		return err
+	}
+	info, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if info.IsDir() {
+		filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+			if info.IsDir() {
+				return nil
+			}
+			ignoredPaths := []string{
+				"/.git/",
+				".sops.yaml",
+			}
+			for _, ignoredPath := range ignoredPaths {
+				if strings.Contains(path, ignoredPath) {
+					return nil
+				}
+			}
+			o := opts
+			o.InputPath = path
+			err = syncFile(o)
+			if err != nil {
+				log.Printf("Failed to sync %s: %s", path, err)
+			}
+			return nil
+		})
+	} else {
+		syncFile(opts)
+	}
+	return nil
+}
+
+func syncFile(opts Opts) error {
```

Comment:
`sync` defines a specific operation (move a state from a source to a destination) that is different from what you're doing here. I'd suggest `update` or `modify`.